### PR TITLE
Update profile-rt and fix LLVM5.0 build

### PR DESCRIPTION
**Pull request by JohanEngelen**
The profile-rt update is a copy from compiler-rt, so nothing to review there.

Only interesting change is in `driver/codegenerator.cpp` (NFC).

### CODE REVIEW

**Comment by JohanEngelen** [comment]
Btw, no interesting changes to profile-rt LLVM 4.0, so no urgent need to cherrypick onto release-1.3.x

**Comment by JohanEngelen** [comment]
@thewilsonator Another LLVM 5.0 update is needed for OSX spirv.

**Comment by thewilsonator** [comment]
Done, sorry for the wait.

File: driver/codegenerator.cpp [hunk 1]
```diff
@@ -61,8 +61,13 @@ createAndSetDiagnosticsOutputFile(IRState &irs, llvm::LLVMContext &ctx,
         llvm::make_unique<llvm::yaml::Output>(diagnosticsOutputFile->os()));
 
     // If there is instrumentation data available, also output function hotness
-    if (!global.params.genInstrProf && global.params.datafileInstrProf)
+    if (!global.params.genInstrProf && global.params.datafileInstrProf) {
+#if LDC_LLVM_VER >= 500
+      ctx.setDiagnosticsHotnessRequested(true);
+#else
       ctx.setDiagnosticHotnessRequested(true);
+#endif
```
Comment (kinke) [inline driver/codegenerator.cpp]: They changed it back...

Comment (JohanEngelen) [inline driver/codegenerator.cpp]: checked: current trunk has the extra "s"

**Pull request merged**